Support for torch.nested.as_nested_tensor(t)

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #121518
* __->__ #113280
* #113279

This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

jbschlosser committed:
Support for torch.nested.as_nested_tensor(t)

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser commented on test/test_nestedtensor.py:
```diff
@@ -3317,6 +3317,59 @@ def test_jagged_view_from_values_offsets(self, device, dtype, requires_grad):
+
+        if torch.device(device) == t.device and dtype == t.dtype and contiguous:
+            # should be the non-copying (view) case
+            # TODO: Figure out why this is broken!
```
nah, good catch. should be able to remove it now

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

voznesenskym committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

voznesenskym committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

voznesenskym committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

voznesenskym committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

voznesenskym committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

voznesenskym committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

Skylion007 commented on torch/nested/__init__.py:
```diff
@@ -52,23 +61,59 @@ def as_nested_tensor(
         )
 
+    if isinstance(ts, Tensor) and ts.dim() < 2:
+        raise RuntimeError("as_nested_tensor(): Expected tensor argument to have dim() > 1")
```
```suggestion
        raise ValueError("as_nested_tensor(): Expected tensor argument to have dim() > 1")
```

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge

jbschlosser commented:
@pytorchbot merge -f "ignore unrelated failures"

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge

jbschlosser commented:
@pytorchbot revert -m "Need to fix BC thing" -c ghfirst

jbschlosser commented:
@pytorchbot revert -m "Need to fix BC thing" -c ghfirst

jbschlosser commented:
@pytorchbot revert -m "Need to fix BC thing" -c ghfirst

jbschlosser committed:
Update on "Support for torch.nested.as_nested_tensor(t)"


This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge
